### PR TITLE
Search experience improvements. Enables new search experience hidden pref.

### PR DESCRIPTION
**Pull request by boek**

### CODE REVIEW

File: app/src/main/java/org/mozilla/fenix/searchdialog/SearchDialogFragment.kt [hunk 1]
```diff
@@ -97,13 +112,41 @@ class SearchDialogFragment : AppCompatDialogFragment() {
...
+                clear(awesomeBar.id, TOP)
+                clear(awesomeBar.id, BOTTOM)
+                connect(awesomeBar.id, TOP, PARENT_ID, TOP)
+                connect(awesomeBar.id, BOTTOM, toolbar.id, TOP)
+
+                applyTo(search_wrapper)
+            }
+        }
+
+        search_wrapper.setOnClickListener {
```
Comment (liuche) [inline app/src/main/java/org/mozilla/fenix/searchdialog/SearchDialogFragment.kt:131]: okay makes sense, if you tap the blank space, dismiss the dialog

File: app/src/main/java/org/mozilla/fenix/home/HomeFragment.kt [hunk 2]
```diff
@@ -383,8 +383,10 @@ class HomeFragment : Fragment() {
             }
         }
 
+        val args by navArgs<HomeFragmentArgs>()
+
         if (view.context.settings().accessibilityServicesEnabled &&
-            bundleArgs.getBoolean(FOCUS_ON_ADDRESS_BAR)
+            args.focusOnAddressBar
```
[the remark this answers is not in the document]
Comment (boek) [reply app/src/main/java/org/mozilla/fenix/home/HomeFragment.kt:389]: Yep! It's a kotlin helper :)

File: app/src/main/res/layout/fragment_search_dialog.xml [hunk 3]
```diff
@@ -25,29 +28,22 @@
         app:layout_constraintEnd_toEndOf="parent"
         app:layout_constraintStart_toStartOf="parent"
         app:layout_constraintTop_toTopOf="parent"/>
-    <androidx.core.widget.NestedScrollView
+    <mozilla.components.browser.awesomebar.BrowserAwesomeBar
```
[the remark this answers is not in the document]
Comment (boek) [reply app/src/main/res/layout/fragment_search_dialog.xml:31]: The awesomebar is a recycclerview (which is it's own scrollview). Now that we added the combined adapters we should be able to get away with just one recyclerview.

File: app/src/main/java/org/mozilla/fenix/searchdialog/SearchDialogFragment.kt [hunk 4]
```diff
@@ -97,6 +103,22 @@ class SearchDialogFragment : AppCompatDialogFragment() {
     override fun onViewCreated(view: View, savedInstanceState: Bundle?) {
         super.onViewCreated(view, savedInstanceState)
 
+        if (view.context.settings().toolbarPosition == ToolbarPosition.BOTTOM) {
```
[the remark this answers is not in the document]
Comment (boek) [reply app/src/main/java/org/mozilla/fenix/searchdialog/SearchDialogFragment.kt:115]: Yep this is how we do it on the home and browser screen!

File: app/src/main/java/org/mozilla/fenix/searchdialog/SearchDialogFragment.kt (hunk 1 above)
Comment (boek) [reply app/src/main/java/org/mozilla/fenix/searchdialog/SearchDialogFragment.kt:131]: I'm not sure if there is a better way to do this, but this is what we ended up doing on the tabs tray as well

**Pull request merged**